xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/utilities/XWikiIcon.vue

```diff
@@ -19,47 +19,113 @@
  -->
 
 <template>
+  <img v-if="isImage" :src="url" :class="additionalClasses"/>
+  <span v-else-if="isFont" :class="[cssClass, ...additionalClasses]"></span>
+  <span v-else class="icon-placeholder"></span>
+</template>
 
-  <img
-    v-if="iconSetType === 'IMAGE'"
-    :src="iconDescriptor.url"
-  />
 
-  <span
-    v-else-if="iconSetType === 'FONT'"
-    :class="iconDescriptor.cssClass"
-  ></span>
+<script>
 
-  <span
-    v-else
-    class="icon-placeholder"
-  ></span>
+import $ from "jquery";
```

Comment:
I agree with you that using fetch is better for the future but we need to evaluate the existing polyfill implementation before choosing one. I see if I can do it before the merge, otherwise I'll create a task for that.

Reply:
I managed to integrate `whatwg-fetch`, `XWikiIcon` does not depend on jQuery anymore and is working on IE11 too.